.github/workflows/build.yml

```diff
@@ -1,111 +1,94 @@
+        type: string
+        default: TestCategory!=Integration
       coverage-threshold:
         required: false
         type: number
         default: 100
-    outputs:
-      beta-version:
-        description: "Beta version number."
-        value: ${{jobs.Pack.outputs.beta-version}}
-      stable-version:
-        description: "Stable version number."
-        value: ${{jobs.Pack.outputs.stable-version}}
 
 jobs:
-  Pack:
+  Build:
     runs-on: ubuntu-latest
-    outputs:
-      beta-version: ${{steps.version.outputs.beta-version}}
-      stable-version: ${{steps.version.outputs.stable-version}}
+    env:
+      OFFICIAL_BUILD: "True"
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-
-    - name: Setup .NET
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
-    - name: Test
-      run: dotnet test /p:CollectCoverage=true /p:Threshold=${{inputs.coverage-threshold}} /p:ThresholdType=line /p:ThresholdStat=total /p:CoverletOutputFormat=cobertura /p:CoverletOutput=$GITHUB_WORKSPACE/coverage.xml
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 6.0.x
 
-    - name: Generate Test Coverage Report
-      uses: danielpalme/ReportGenerator-GitHub-Action@5.1.3
-      with:
-        reports: 'coverage.xml'
-        targetdir: 'coveragereport'
-        reporttypes: 'HtmlInline;Cobertura;Badges' # The output formats and scope (separated by semicolon) Values: Badges, Clover, Cobertura, CsvSummary, Html, Html_Dark, Html_Light, HtmlChart, HtmlInline, HtmlInline_AzurePipelines, HtmlInline_AzurePipelines_Dark, HtmlInline_AzurePipelines_Light, HtmlSummary, JsonSummary, Latex, LatexSummary, lcov, MarkdownSummary, MHtml, PngChart, SonarQube, TeamCitySummary, TextSummary, Xml, XmlSummary
-        verbosity: 'Info'
-        tag: '${{ github.run_number }}_${{ github.run_id }}'
-        toolpath: 'reportgeneratortool'
-      if: always()
+      - name: Restore packages
+        run: dotnet restore
 
-    - name: Upload Coverage Report Artifact
-      uses: actions/upload-artifact@v2
-      with:
-        name: CoverageReport   
-        path: coveragereport
-      if: always()
+      - name: Build
+        run: dotnet build "${{inputs.project-to-build}}" --no-restore --configuration Debug
 
-    - name: Install GitVersion
-      uses: gittools/actions/gitversion/setup@v0.9.7
-      with:
-        versionSpec: '5.x'
-
-    - name: Determine Version
-      id:   gitversion
-      uses: gittools/actions/gitversion/execute@v0.9.7
-      with:
-        useConfigFile: true
-
-    - name: Set version variables
-      run: |
-        echo "STABLE=${{steps.gitversion.outputs.majorMinorPatch}}" >> $GITHUB_ENV
-        echo "BETA=${{steps.gitversion.outputs.majorMinorPatch}}-beta.$(date +%s)" >> $GITHUB_ENV
+  Test:
```

Comment:
why is Test a separate job? why can't we re-use the same VM provisioned for the Build step and make them one job

Reply:
Will do